### PR TITLE
Refactor the `dhcp_update_required?` method

### PR DESCRIPTION
**Pull request by archanaserver**
Based on discussion: https://github.com/theforeman/foreman/pull/10068/files/5dc13229ccc9e02791121f055b7a7b7d4a14918e#r1699753281

Extracted complex condition into separate methods and handle the TODO comment related to jumpstart logic.

### CODE REVIEW

File: app/models/concerns/orchestration/dhcp.rb [hunk 1]
```diff
@@ -171,17 +171,18 @@ def dhcp_update_required?
...
       (!old.build? && build? && !all_dhcp_records_valid?))
-    # Handle jumpstart
-    # TODO, abstract this way once interfaces are fully used
-    if is_a?(Host::Base) && jumpstart?
-      if !old.build? || (old.medium != medium || old.arch != arch) ||
-          (os && old.os && (old.os.name != os.name || old.os != os))
-        return true
-      end
+
+    if requires_jumpstart_update?
```
Comment (archanaserver) [inline app/models/concerns/orchestration/dhcp.rb:175]: can we refactor it more like:

```
return true if requires_jumpstart_update?
  false
```

**Review by ekohl — CHANGES_REQUESTED** [review]
What I meant was that perhaps it already is properly abstracted away.

The whole method tries to determine if the DHCP record needs to be updated. That is if any parameter has changed. We `Net::DHCP::Record` and subclasses in [lib/net/dhcp](https://github.com/theforeman/foreman/tree/develop/lib/net/dhcp).

These are called in `ProxyAPI::DHCP`:
https://github.com/theforeman/foreman/blob/2f947f7f31f960f8b297a764d4b4d070b1b250f8/app/services/proxy_api/dhcp.rb#L41-L57

And indirectly in orchestration:
https://github.com/theforeman/foreman/blob/2f947f7f31f960f8b297a764d4b4d070b1b250f8/app/models/concerns/orchestration/dhcp.rb#L101-L109

The latter one is more relevant (it's also this file), so `dhcp_update_required?` should compare all attributes that `dhcp_attrs` can return.

Note it has this bit already:
https://github.com/theforeman/foreman/blob/2f947f7f31f960f8b297a764d4b4d070b1b250f8/app/models/concerns/orchestration/dhcp.rb#L129-L130

So I'd expect you can compare `os.jumpstart_params` and `old.os.jumpstart_params`.

File: app/models/concerns/orchestration/dhcp.rb (hunk 1 above)
Comment (adamruzicka) [reply app/models/concerns/orchestration/dhcp.rb:175]: Wouldn't even just `requires_jumpstart_update?` do the trick? Or double-negate it to cast it from a truthy/falsey value into a bool?

**Review by adamruzicka — COMMENTED** [review]
Just to be extra sure I'm not missing anything, this just extracts a chunk of code into a method, correct? There are no actual changes to what are all the things that get checked, right?

Edit: somehow I haven't seen Ewoud's previous comment, oh well

**Pull request closed without merging**